skimage/morphology/_misc_cy.pyx

```diff
@@ -0,0 +1,122 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+
+"""Cython code used in `remove_near_objects` function."""
+
+cimport numpy as cnp
+
+from .._shared.fused_numerics cimport np_anyint
+
+
+def _remove_near_objects(
+    np_anyint[::1] out not None,
+    Py_ssize_t[::1] border_indices not None,
+    Py_ssize_t[::1] inner_indices not None,
+    kdtree,
+    cnp.float64_t p_norm,
+    cnp.float64_t min_distance,
+    tuple shape,
+):
+    """Remove objects until a minimal distance is ensured.
+
+    Iterates over all objects (pixels that aren't zero) inside an image and
+    removes "nearby" objects until all remaining ones are spaced more than a
+    given minimal distance from each other.
+
+    Parameters
+    ----------
+    out :
```

Comment:
type not given?

Reply:
Also, why is "out" the first parameter?

Reply:
Sorry, I just closed this because @stefanv realized later on that this was an internal function during review (which he commented). Nevertheless, I should have given an update anyway. 

The type is not given, because this is a typed Cython function already and I didn't want them to be redundant and potentially out of date.

"out" is first, because it made sense to me. It's the only value that is modified and all other parameters are just there to help with that.